auth-foundation/src/test/java/com/okta/authfoundation/client/NetworkUtilsTest.kt

```diff
@@ -694,61 +694,90 @@ class NetworkingTest {
 
-        val startTime = currentTime
         val result = oktaRule.createOidcClient().performRequest(JsonObject.serializer(), request) {
             it["foo"]!!.jsonPrimitive.content
         }
-        val elapsedTime = currentTime - startTime
 
-        assertThat((result as OidcClientResult.Success<String>).result).isEqualTo("bar")
-        assertThat(elapsedTime).isEqualTo(minDelaySeconds.seconds.inWholeMilliseconds)
+        val exception = (result as OidcClientResult.Error<String>).exception
+        assertThat(exception).isInstanceOf(OidcClientResult.Error.HttpResponseException::class.java)
+        val httpResponseException = exception as OidcClientResult.Error.HttpResponseException
+        assertThat(httpResponseException.responseCode).isEqualTo(429)
+
+        assertThat(oktaRule.mockWebServerDispatcher.numberRemainingInQueue()).isEqualTo(1)
+        oktaRule.mockWebServerDispatcher.clear()
     }
 
     @Test fun testPerformRequestRetryWithInvalidRateLimitResetHeaderShouldWaitMinDelay(): Unit = runTest {
-        val minDelaySeconds = 20L
-        val rateLimitEventHandler = getRateLimitExceededEventHandler(minDelaySeconds = minDelaySeconds)
-        oktaRule.eventHandler.registerEventHandler(rateLimitEventHandler)
-
         oktaRule.enqueue(path("/test")) { response ->
             response.setResponseCode(429)
             setRateLimitHeaders(response)
             response.setHeader("x-rate-limit-reset", "abcde")
             response.setBody("""{"foo":"bar"}""")
         }
-        oktaRule.enqueue(path("/test")) { response ->
-            setRateLimitHeaders(response)
-            response.setBody("""{"foo":"bar"}""")
+        oktaRule.enqueue(path("/test")) { _ ->
+            throw IllegalStateException("Retried request after getting invalid 429 x-rate-limit-reset header.")
         }
         val request = Request.Builder()
             .url(oktaRule.baseUrl.newBuilder().addPathSegments("test").build())
             .build()
 
-        val startTime = currentTime
         val result = oktaRule.createOidcClient().performRequest(JsonObject.serializer(), request) {
             it["foo"]!!.jsonPrimitive.content
         }
-        val elapsedTime = currentTime - startTime
 
-        assertThat((result as OidcClientResult.Success<String>).result).isEqualTo("bar")
-        assertThat(elapsedTime).isEqualTo(minDelaySeconds.seconds.inWholeMilliseconds)
+        val exception = (result as OidcClientResult.Error<String>).exception
+        assertThat(exception).isInstanceOf(OidcClientResult.Error.HttpResponseException::class.java)
+        val httpResponseException = exception as OidcClientResult.Error.HttpResponseException
+        assertThat(httpResponseException.responseCode).isEqualTo(429)
+
+        assertThat(oktaRule.mockWebServerDispatcher.numberRemainingInQueue()).isEqualTo(1)
+        oktaRule.mockWebServerDispatcher.clear()
+    }
+
+    @Test fun testPerformRequestRetrySendsAnEventForEach429Response(): Unit = runTest {
```

Comment:
We should also make sure the most recent event is the one that's used. So we should change something about it and validate it.